app/models/team.rb

```diff
@@ -535,7 +535,11 @@ def filtered_fact_checks(filters = {})
     query = query.where('fact_checks.report_status' => filters[:report_status].to_a.map(&:to_s)) unless filters[:report_status].blank?
 
     # Filter by text
-    query = query.where('(fact_checks.title ILIKE ? OR fact_checks.url ILIKE ? OR fact_checks.summary ILIKE ?)', *["%#{filters[:text]}%"]*3) if filters[:text].to_s.size > 2
+    if filters[:text].to_s.size > 2
+      tsquery = Team.sanitize_sql_array(["to_tsquery(?)", filters[:text].split(/\s+/).map(&:strip).join(' & ')]) # FIXME: May not work for all languages
```

Comment:
Not sure if `plainto_tsquery` is any better for other languages? It does seem simpler, however as we don't have to split the query: https://www.postgresql.org/docs/current/textsearch-controls.html

Reply:
Or maybe even better is `websearch_to_tsquery`:
> websearch_to_tsquery creates a tsquery value from querytext using an alternative syntax in which simple unformatted text is a valid query. Unlike plainto_tsquery and phraseto_tsquery, it also recognizes certain operators. Moreover, this function will never raise syntax errors, which makes it possible to use raw user-supplied input for search. The following syntax is supported:

Reply:
yes @computermacgyver `websearch_to_tsquery`  is better I used it and verified that everything works as expected

Reply:
@melsawy , since we're using `websearch_to_tsquery`, do we still need to `split` the query? It's better if we don't need to.